src/controllers/completion/sources/UserSource.cpp

```diff
@@ -58,7 +58,16 @@ void UserSource::initializeFromChannel(const Channel *channel)
         return;
     }
 
-    this->items_ = tc->accessChatters()->all();
+    if (getSettings()->alwaysIncludeBroadcasterInMentions)
+    {
+        this->items_ = tc->accessChatters()->allExcept(tc->getName());
+        this->items_.insert(this->items_.begin(),
+                            {tc->getName(), tc->getDisplayName()});
+    }
+    else
+    {
+        this->items_ = tc->accessChatters()->all();
+    }
```

Comment:
@pajlada I wasn't able to add the suggestion to the other comment. How does this look?

```suggestion
    if (getSettings()->alwaysIncludeBroadcasterInMentions)
    {
        auto broadcaster = tc->getName();
        auto it = std::find_if(this->items_.begin(), this->items_.end(),
                               [&broadcaster](const UserItem &user) {
                                   return user.first == broadcaster;
                               });
        if (it != this->items_.end())
        {
            std::rotate(it, it + 1, this->items_.begin());
        }
        else
        {
            this->items_.insert(this->items_.begin(),
                                {tc->getName(), tc->getDisplayName()});
        }
    }
```

Reply:
That looks reasonable at first glance to me, commit that & I will test & review fully this weekend

Reply:
The `std::rotate` will break the order of the completions (the last chatter won't be first). Not sure if people rely on that.

Reply:
The rotate logic also throws when the streamer is already at the last position in the vector. I'll fix this logic (maybe will have to erase/insert) and add some unit tests later

Reply:
> [a, b, c, streamer, e] would become [streamer, a, b, c, e]

That would become [streamer, e, a, b, c].

[Cppreference](https://en.cppreference.com/w/cpp/algorithm/rotate) has further explanation on this.

You can do this in one loop using `emplace` and `erase` I believe:

```cpp
// auto items = ...;
// auto streamer = {streamerLogin, streamerName};
auto it = items.emplace(items.begin(), streamer);
while(it != items.end()) {
  if (it->first == streamerLogin) {
    it = items.erase(it);
  } else {
    it++;
  }
}
```